rmsutil/crypto.py

```diff
@@ -0,0 +1,41 @@
+import base64
+import hashlib
+import hmac
+
+
+def opaque_id(key, identifier):
+    """Calculate an opaque ID corresponding to a private identifier.
+
+    This function uses HMAC-SHA256 to hash a private identifier to an
+    unpredictable, opaque ID. The hashed ID can be used in place of the
+    original in order to avoid exposing a potentially sensitive value,
+    such as a database table's monotonic primary key, in an insecure
+    context.
+
+    Args:
+        key (six.text_type): URL-safe Base64-encoded HMAC key, at least
+            16 bytes in length.
```

Comment:
I would say "128 bits of entropy" but that may not be particularly enlightening to the end-user :)